src/viewport_manager.ts

```diff
@@ -64,9 +70,6 @@ export class ViewportManager implements Disposable {
         col,
         topline,
     }: EventBusData<"viewport-changed">[0]) {
-        this.viewportChangedPromise?.resolve();
-        this.viewportChangedPromise = new ManualPromise();
```

Comment:
Is there a reason you moved these outside of the function? If the issue is simply the `gridId` return causing the promise to be left as an unresolved `ManualPromise`, I think you could simplify this by doing

```ts
this.viewportChangedPromise?.resolve();
this.viewportChangedPromise = undefined;


const gridId = this.main.bufferManager.getGridIdForWinId(winid);
if (!gridId) {
  logger.warn(`Unable to update scrolled view. No grid for winId: ${winid}`);
  return;
}

this.viewportChangedPromise = new ManualPromise();
// ...
```

or am I missing something?